src/Form/EventListener/UseInputOptionsAsEventDataEventSubscriber.php

```diff
@@ -39,9 +39,15 @@ private function convertInputToSubmittedData(InputInterface $input, FormInterfac
                     $submittedData = $this->convertInputToSubmittedData($input, $field, $name ?? $form->getName());
                 } else {
                     $subName = $name === null ? $childName : $name . '[' . $childName . ']';
-                    $submittedData[$childName] = $this->convertInputToSubmittedData($input, $field, $subName);
+                    $subValue = $this->convertInputToSubmittedData($input, $field, $subName);
+                    if ($subValue !== null) {
+                        $submittedData[$childName] = $subValue;
+                    }
                 }
             }
+            if (empty($submittedData)) {
```

Comment:
Could this be more specific? E.g. `$submittedData === []`?

Reply:
Sure!